wgpu-core/src/instance.rs

```diff
@@ -217,6 +218,30 @@ impl AdapterInfo {
     }
 }
 
+#[derive(Clone, Debug, PartialEq)]
+/// Error when requesting a device from the adaptor
+pub enum RequestDeviceError {
+    /// Unsupported feature extension was requested
+    UnsupportedFeature(wgt::Features),
+    /// Requested device limits were exceeded
+    LimitsExceeded,
+}
+
+impl Display for RequestDeviceError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match &self {
+            RequestDeviceError::UnsupportedFeature(features) => write!(
+                f,
+                "Cannot enable features that adapter doesn't support. Unsupported extensions: {:?}",
+                features
+            ),
+            RequestDeviceError::LimitsExceeded => {
+                write!(f, "Adapter does not support the requested max_bind_groups",)
```

Comment:
comment still needs to be updated, it should tell that *some* of the limits are exceeding

Reply:
Sounds good, I made more general